tests/test_extras.py

```diff
@@ -1,24 +1,19 @@
+import pkgutil
 from importlib import import_module
-from pathlib import Path
 from typing import List
 
 import pytest
 
-PATH_TO_EXTRAS = "streamlit_extras"
+import streamlit_extras
 
 
 def get_extras() -> List[str]:
-    extra_names = [
-        folder.name
-        for folder in Path(PATH_TO_EXTRAS).glob("*")
-        if folder.is_dir() and folder.name != "__pycache__"
-    ]
-    return extra_names
+    return [extra.name for extra in pkgutil.iter_modules(streamlit_extras.__path__)]
 
 
 @pytest.mark.parametrize("extra", get_extras())
 def test_extra_attributes(extra: str):
-    mod = import_module(f"{PATH_TO_EXTRAS}.{extra}")
+    mod = import_module(f"streamlit_extras.{extra}")
```

Comment:
I think because `streamlit-extras` is not a requirement in poetry, this could break? Hence the usage of the directory traversal instead of the package...

Reply:
It will only break if you don't have the package installed, which moving it to `src` encourages you more to do. And, the default behavior of `poetry install`, which does happen in the github action, is to install the dependencies and any `packages` that you specify in pyproject.toml